src/lighteval/models/vllm_model.py

```diff
@@ -0,0 +1,362 @@
+            trust_remote_code (bool, optional): Whether to trust remote code. Defaults to False.
+            cache_dir (str, optional): The cache directory for the model. Defaults to "/scratch".
+
+        Returns:
+            transformers.PreTrainedModel: The created auto model instance.
+        """
+        self.model_args = {
+            "model": config.pretrained,
+            "gpu_memory_utilization": float(0.8),
+            "revision": config.revision + (f"/{config.subfolder}" if config.subfolder is not None else ""),
+            "dtype": config.dtype,
+            "trust_remote_code": config.trust_remote_code,
+            "tensor_parallel_size": int(1),
+            "max_model_len": int(self._max_length) if self._max_length else None,
+            "swap_space": 4,
+            "seed": 1234,
+        }
+        if int(config.data_parallel_size) > 1:
+            self.model_args["worker_use_ray"] = True
+            self._batch_size = "auto"
+            return None
+
+        model = LLM(**self.model_args)
+        return model
+
+    def _create_auto_tokenizer(self, config: VLLMModelConfig, env_config: EnvConfig):
+        tokenizer = get_tokenizer(
+            config.pretrained,
+            tokenizer_mode="auto",
+            trust_remote_code=config.trust_remote_code,
+            tokenizer_revision=config.revision,
+        )
+        tokenizer.pad_token = tokenizer.eos_token
+        return tokenizer
+
+    def _init_max_length(self, max_length) -> int:
+        """Return the maximum sequence length of the model.
+        NOTE: Different model configurations have different max sequence length
+        attribute names.
+            - n_positions: (CTRLConfig)
+            - max_position_embeddings: (BartConfig, RoFormerConfig)
+            - n_ctx: (GPT2Config)
+        NOTE: For relative position encoded models you should specify the max
+        sequence length of the model in the constructor via `max_length`.
+
+        Args:
+            max_length (Optional[int]): The maximum length of the input sequence. If not provided, it will be determined
+                based on the model's configuration or tokenizer's model_max_length attribute.
+
+        Returns:
+            int: Max length to use depending on the available args and config
+        """
+        if max_length is not None:
+            return int(max_length)
+        # Try to get the sequence length from the model config.
+        seqlen_config_attrs = ("n_positions", "max_position_embeddings", "n_ctx")
+
+        for attr in seqlen_config_attrs:
+            if hasattr(self._config, attr):
+                return getattr(self._config, attr)
+
+        # Default max sequence length setting for when no `max_length` is provided
+        # or no max length config setting is found in the model or tokenizer.
+        return 2048
+
+    def greedy_until(
```

Comment:
How much of this code overlaps with the HF code? Could we make it inherit from the base model?

Reply:
there isn't much overlap as we needed to rewrite the greedy_untill and loglikehood funciton and most of the other function from base are not needed.

Reply:
the other functions are rewritten too and it makes the code much simpler. I would like to not depend on `base_model` so that a refacto of the model APIs will be easier in the future